test/src/timeout_tests.erl

```diff
@@ -0,0 +1,131 @@
+    {_Writer, Limiter, Ch} = test_channel(),
+    [Q, Q2] = [Queue || Name <- [<<"foo">>, <<"bar">>],
+                        {new, Queue = #amqqueue{}} <-
+                            [rabbit_amqqueue:declare(
+                               rabbit_misc:r(<<"/">>, queue, Name),
+                               false, false, [], none)]],
+    ok = rabbit_amqqueue:basic_consume(
+           Q, true, Ch, Limiter, false, 0, <<"ctag">>, true, [], undefined),
+
+    %% list users
+    ok = control_action(add_user, ["foo", "bar"]),
+    {error, {user_already_exists, _}} =
+        control_action(add_user, ["foo", "bar"]),
+    ok = control_action(list_users, [], ?TIMEOUT_LIST_OPS_PASS),
+    
+    %% list parameters
+    ok = rabbit_runtime_parameters_test:register(),
+    ok = control_action(set_parameter, ["test", "good", "123"]),
+    ok = control_action(list_parameters, [], ?TIMEOUT_LIST_OPS_PASS),
+    ok = control_action(clear_parameter, ["test", "good"]),
+    rabbit_runtime_parameters_test:unregister(),
+
+    %% list vhosts
+    ok = control_action(add_vhost, ["/testhost"]),
+    {error, {vhost_already_exists, _}} =
+        control_action(add_vhost, ["/testhost"]),
+    ok = control_action(list_vhosts, [], ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list permissions
+    ok = control_action(set_permissions, ["foo", ".*", ".*", ".*"],
+                        [{"-p", "/testhost"}]),
+    ok = control_action(list_permissions, [], [{"-p", "/testhost"}],
+                        ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list user permissions
+    ok = control_action(list_user_permissions, ["foo"],
+                        ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list policies
+    ok = control_action_opts(["set_policy", "name", ".*",
+                              "{\"ha-mode\":\"all\"}"]),
+    ok = control_action(list_policies, [], ?TIMEOUT_LIST_OPS_PASS),
+    ok = control_action(clear_policy, ["name"]),
+
+    %% list queues
+    ok = info_action(list_queues, rabbit_amqqueue:info_keys(), false,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list exchanges
+    ok = info_action(list_exchanges, rabbit_exchange:info_keys(), true,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list bindings
+    ok = info_action(list_bindings, rabbit_binding:info_keys(), true,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list connections
+    {H, P} = find_listener(),
+    {ok, C} = gen_tcp:connect(H, P, [binary, {active, false}]),
+    gen_tcp:send(C, <<"AMQP", 0, 0, 9, 1>>),
+    {ok, <<1,0,0>>} = gen_tcp:recv(C, 3, 100),
+    ok = info_action(list_connections,
+                     rabbit_networking:connection_info_keys(),
+		     false,
+                     ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list consumers
+    ok = info_action(list_consumers, rabbit_amqqueue:consumer_info_keys(),
+                     false, ?TIMEOUT_LIST_OPS_PASS),
+
+    %% list channels
+    ok = info_action(list_channels, rabbit_channel:info_keys(), false,
```

Comment:
Can you make the tests more thorough? Right now this is just testing that `info_action` runs and that's all. There's a bug right now were channels `info_all/3` only returns the first channel on the list and nothing else. So I would expect first that the code above creates more than one channel, more than one queue, and so on. 

Then instead of asserting that the function `info_action` returns OK, I would assert that the channels I created above are actually listed when I call `list_channels`, just to name an example. The test should be automatic, I shouldn't need to read the screen in order to see if it passed.

Perhaps you have to call this function instead: https://github.com/rabbitmq/rabbitmq-server/pull/303/files#diff-0791814e0c4904b21e083a7378194669R773 in order to make your tests become the aggregator Pid or see how to mock that.